lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -367,13 +365,11 @@ def __init__(
         )
 
         if refresh_events:
-            for ev in refresh_events:
-                self.framework.observe(ev, self._on_refresh_event)
+            logger.warn(
+                "DEPRECATION WARNING. `refresh_events` is now deprecated. CertHandler will automatically refresh the CSR when necessary."
+            )
 
-    def _on_refresh_event(self, _):
-        """Replace the latest current CSR with a new one if there are any SANs changes."""
-        if self._stored.csr_hash != self._csr_hash:
-            self._generate_csr(renew=True)
+        self._refresh_csr_if_needed()
```

Comment:
since that seems to be the pattern, how about we call `self._reconcile` from init, and in `_reconcile` we call this method?
that way it'll be easier to extend when (not if) we want to add more stuff to the common exit hook